### PR TITLE
vlas/fixup setup

### PR DESCRIPTION
**Pull request by vlasaki**
https://mysten.atlassian.net/browse/SE-704

Fixup publish script.
- update testnet address.
- remove gas budget from publish command.
- fix broken command line options --env, --address.
- script to bail on error. it currently prints success message.
- add help message on incorrect command line args.
etc

Please look for test evidence in the comment section of the linked jira.

TODO: 
file app/.env currently has the following in it:
NEXT_PUBLIC_SUI_NETWORK=https://rpc.testnet.sui.io:443
NEXT_PUBLIC_PACKAGE_ADDRESS=0x8f4ba9c068cc90c1140d7310ca81b8db1f43ec38cfb252ec227ac307d8829c8d
NEXT_PUBLIC_ADMIN_ADDRESS=0x65391674eb4210940ea98ae451237d9335920297e7c8abaeb7e05b221ee36917
NEXT_PUBLIC_HOUSE_DATA_ID=0x5e905fd8dcc04cdae722bddfcf8d9ebc07be3fcbe3deb5ebb011dd5f3fba62d0
NEXT_PUBLIC_ENOKI_API_KEY=enoki_public_0ca87f8adcb98686bcaf5798735f1b00
NEXT_PUBLIC_GOOGLE_CLIENT_ID=354902366712-1r87r3a0klkch017ogef3j501dkn3s6m.apps.googleusercontent.com

Myst update the following env variables:
+NEXT_PUBLIC_SUI_NETWORK=https://fullnode.testnet.sui.io:443
+NEXT_PUBLIC_PACKAGE_ADDRESS=0xed8d48f5ef6f86ebca92db38df151c497cb4125fbc2a7b301c34f5039be69ad6
+NEXT_PUBLIC_ADMIN_ADDRESS=0xad9396b530f9fcdaee7dc5bb62d423241caf3426d5e3d937da3e2503fb656f9e
+NEXT_PUBLIC_HOUSE_DATA_ID=0xe90cfa74d7041d4ba495f8e6e39ace88ba2d4938c8144b22394170797ed8f06a

But what about enoki_api_key and google_client_id?

### CODE REVIEW

File: setup/publish.sh [hunk 1]
```diff
@@ -1,72 +1,177 @@
...
+echo "PACKAGE_ADDRESS=$PACKAGE_ID"
+echo "ADMIN_ADDRESS=$ADMIN_ADDRESS"
+echo "ADMIN_SECRET_KEY=$ADMIN_SECRET_KEY"
+echo "BJ_PLAYER_SECRET_KEY=$BJ_PLAYER_SECRET_KEY"
+echo "HOUSE_ADMIN_CAP=$HOUSE_ADMIN_CAP"
 cat >.env<<-API_ENV
 SUI_NETWORK=$NETWORK
 PACKAGE_ADDRESS=$PACKAGE_ID
 ADMIN_ADDRESS=$ADMIN_ADDRESS
+ADMIN_SECRET_KEY=$ADMIN_SECRET_KEY
```
Comment (John-Atha) [inline setup/publish.sh:162]: In general some devs might want to expose the secret keys as env variables only in their terminal, and not be written in files.
I believe we should avoid automatically writing the secret keys to files

File: setup/publish.sh [hunk 2]
```diff
@@ -1,72 +1,177 @@
...
+PUBLISHED_OBJS=$(echo "$PUBLISH_RES" | jq -r '.objectChanges[] | select(.type == "published")')
+PACKAGE_ID=$(echo "$PUBLISHED_OBJS" | jq -r '.packageId')
+CREATED_OBJS=$(echo "$PUBLISH_RES" | jq -r '.objectChanges[] | select(.type == "created")')
+HOUSE_ADMIN_CAP=$(echo "$CREATED_OBJS" | jq -r 'select (.objectType | contains("single_player_blackjack::HouseAdminCap")).objectId')
+
+echo "Publish new env var to setup/.env: "
+echo "SUI_NETWORK=$NETWORK"
+echo "PACKAGE_ADDRESS=$PACKAGE_ID"
+echo "ADMIN_ADDRESS=$ADMIN_ADDRESS"
+echo "ADMIN_SECRET_KEY=$ADMIN_SECRET_KEY"
```
Comment (John-Atha) [inline setup/publish.sh:155]: I believe we should avoid printing the secret keys.
Same thinking as in the following comment

**Review by John-Atha — CHANGES_REQUESTED** [review]
LGTM with a first view, will run it locally after the small fixes related to the secret keys, and will approve!

**Review by John-Atha — APPROVED** [review]
LGTM!

File: setup/publish.sh (hunk 1 above)
Comment (vlasaki) [reply setup/publish.sh]: this is resolved with latest commit

File: setup/publish.sh (hunk 2 above)
Comment (vlasaki) [reply setup/publish.sh]: resolved in latest commit

**Pull request merged**